src/text_extraction/text_extractor.py

```diff
@@ -12,22 +13,47 @@ class TextExtractor():
     def __init__(self):
         self.out_dir = ""
         self.dpi = 500
+        self.metadata_handler = MetadataHandler()
 
-    def read(self, input_file):
+    def read(self, input_file, file_name=None, index=None, uploader=None):
         """ Inner function that reads images and outputs the OCR text"""
-
         out_path = f"{self.out_dir}{re.sub(r'[^.]+$', 'txt', os.path.basename(input_file))}"
 
         text = str(((pytesseract.image_to_string(Image.open(input_file),lang="dan"))))
 
         print("Reading file" + input_file)
 
         text = text.replace("-\n", "")
+        title_match = re.search(r'^[^\n]+', text)
+        title = title_match.group(0) if title_match else "No Title Found"
+
+        metadata_dict = {
+            "File Name": file_name,
+            "Uploader": uploader,
+            "Index": index,
+            "Title": title,
+            # Add other metadata as needed
+        }
 
         cleaned_text = clean_sentence(text)
 
         # Save each sentence as a new line in the output file
         with open(out_path, 'w', encoding='utf-8') as file:
+            # Use MetadataHandler to write metadata
+            self.metadata_handler.write_metadata(file, metadata_dict)
+
+            sentences = re.split(r'(?<!\w\.\w.)(?<![A-Z][a-z]\.)(?<=\.|\?)\s', text)
+
+            sentence_index = 1
+
+            for _, sentence in enumerate(sentences):
```

Comment:
for i, sentence in enumerate(sentences, 1):

Ændrer start index til 1 og så behøves ekstra kode ikke